packages/app/src/cli/services/function/common.ts

```diff
@@ -30,10 +32,20 @@ export async function inFunctionContext({
   const specifications = await loadLocalExtensionsSpecifications()
   const app: AppInterface = await loadApp({specifications, directory: path, userProvidedConfigName})
 
-  const allFunctions = app.allExtensions.filter((ext) => ext.isFunctionExtension)
-  const ourFunction = allFunctions.find((fun) => fun.directory === path) as ExtensionInstance<FunctionConfigType>
+  const allFunctions = app.allExtensions.filter(
+    (ext) => ext.isFunctionExtension,
+  ) as ExtensionInstance<FunctionConfigType>[]
+  const ourFunction = allFunctions.find((fun) => fun.directory === path)
+
   if (ourFunction) {
     return callback(app, ourFunction)
+  } else if (isTerminalInteractive()) {
+    const selectedFunction = await renderAutocompletePrompt({
+      message: 'Which function would you like to run the command on?',
```

Comment:
This feels a bit verbose. Thinking about this as a "conversation" with the user, how about just "Which function?"

> `shopify app function build`
>
> ? Which function?

Reply:
Let's go with "Which function?"